### PR TITLE
Update micro-http for empty MMDS path handling

### PR DESCRIPTION
**Pull request by zulinx86**
## Changes

- Check the return code of command run on guests.
- Update Cargo.lock to use the latest micro-http.
- Add an integration test for an empty MMDS path.

## Reason

When querying a MMDS path that has an empty content, the response didn't
contain 'Content-Length' field, resulting in the client waiting for the
connection closed by the server or timeout. To fix this issue,
micro-http made a change to have the field when the response status code
is not 1XX or 204.

## License Acceptance

By submitting this pull request, I confirm that my contribution is made under
the terms of the Apache 2.0 license. For more information on following Developer
Certificate of Origin and signing off your commits, please check
[`CONTRIBUTING.md`][3].

## PR Checklist

- ~~[ ] If a specific issue led to this PR, this PR closes the issue.~~
- [x] The description of changes is clear and encompassing.
- ~~[ ] Any required documentation changes (code and docs) are included in this
  PR.~~
- ~~[ ] API changes follow the [Runbook for Firecracker API changes][2].~~
- [x] User-facing changes are mentioned in `CHANGELOG.md`.
- [x] All added/changed functionality is tested.
- ~~[ ] New `TODO`s link to an issue.~~
- [x] Commits meet
  [contribution quality standards](https://github.com/firecracker-microvm/firecracker/blob/main/CONTRIBUTING.md#contribution-quality-standards).

______________________________________________________________________

- [x] This functionality cannot be added in [`rust-vmm`][1].

[1]: https://github.com/rust-vmm
[2]: https://github.com/firecracker-microvm/firecracker/blob/main/docs/api-change-runbook.md
[3]: https://github.com/firecracker-microvm/firecracker/blob/main/CONTRIBUTING.md

### CODE REVIEW

**Review by roypat — APPROVED** [review]

File: CHANGELOG.md [hunk 1]
```diff
@@ -74,6 +74,10 @@ and this project adheres to
   mechanism to reliably fetch Firecracker PID. With this change, Firecracker
   process's PID will always be available in the Jailer's root directory
   regardless of whether new_pid_ns was set.
+- [#4468](https://github.com/firecracker-microvm/firecracker/pull/4468): Made
```
Comment (pb8o) [inline CHANGELOG.md:77]: Nit: maybe refer to the micro-http commit/PR in the commit message.

File: CHANGELOG.md [hunk 2]
```diff
@@ -74,6 +74,10 @@ and this project adheres to
...
+  MMDS return a response including 'Content-Length' header field even if the
+  body is empty. Otherwise, the client may keep waiting for the connection
```
Comment (pb8o) [inline CHANGELOG.md:79]: nit: I would rewrite as: "Fixed a bug where an MMDS client would hang or timeout when querying for an empty field, because the `Content-Length` response header was missing.

**Review by pb8o — DISMISSED** [review]

File: CHANGELOG.md (hunk 1 above)
Comment (zulinx86) [reply CHANGELOG.md]: added the PR link in the commit message! thanks!

File: CHANGELOG.md (hunk 2 above)
Comment (zulinx86) [reply CHANGELOG.md]: thanks! applied your suggestion!

**Review by roypat — APPROVED** [review]

**Review by pb8o — APPROVED** [review]
